packages/airnode-utilities/src/evm/gas-prices/gas-oracle.ts

```diff
@@ -104,7 +129,10 @@ export const fetchProviderRecommendedGasPrice = async (
   };
 };
 
-export const fetchBaseFeePerGas = async (provider: Provider, startTime: number): Promise<ethers.BigNumber | null> => {
+export const fetchBaseFeePerGas = async (
+  provider: Provider,
+  startTime: number
+): Promise<ethers.providers.Block['baseFeePerGas']> => {
   const goLatestBlock = await go(() => provider.getBlock('latest'), {
```

Comment:
This function throws error when block header couldn't be fetched. In this case, gasOracle tries to the following strategy if it is defined. In case it's defined, it'll be probably `providerRecommendedGasPrice` strategy and again we make another RPC request to get `providerRecommendedGasPrice`. Or the other case where there is no next strategy defined, gasOracle will use constant gas price which is not safe for all chain. So instead of throwing error when block header couldn't be fetched, returning already fetched and calculated providerRecommendedGasPrice is more efficient and safe way to go.

Reply:
Or better and more elegant solution is first making RPC call for block header (throws error if fail) and then making RPC call for `providerRecommendedGasPrice`. So in this case, if block header can't be fetched, gasOracle tries the next strategy which will be probably `providerRecommendedGasPrice`.